src/Stratis.Bitcoin.Tests/P2P/NetworkPeerServerTests.cs

```diff
@@ -51,16 +54,35 @@ public void Validate_AllowClientConnection_State(bool inIBD, bool isWhiteListed,
 
             var asyncProvider = this.CreateAsyncProvider();
 
+            var peerAddressManager = new Mock<IPeerAddressManager>();
+            peerAddressManager.Setup(pam => pam.FindPeersByIp(It.IsAny<IPEndPoint>())).Returns(new List<PeerAddress>());
+
             var networkPeerServer = new NetworkPeerServer(this.Network,
-                endpointAddNode, endpointAddNode, ProtocolVersion.PROTOCOL_VERSION, this.extendedLoggerFactory,
-                networkPeerFactory.Object, initialBlockDownloadState.Object, connectionManagerSettings, asyncProvider);
+                endpointAddNode,
+                endpointAddNode,
+                ProtocolVersion.PROTOCOL_VERSION,
+                this.extendedLoggerFactory,
+                networkPeerFactory.Object,
+                initialBlockDownloadState.Object,
+                connectionManagerSettings,
+                asyncProvider,
+                peerAddressManager.Object,
+                DateTimeProvider.Default);
 
             // Mimic external client
             const int portNumber = 80;
             var client = new TcpClient("www.stratisplatform.com", portNumber);
 
+            string ip = string.Empty;
             var ipandport = client.Client.RemoteEndPoint.ToString();
-            var ip = ipandport.Replace(ipandport.Substring(ipandport.IndexOf(':')), "");
+            if (client.Client.RemoteEndPoint.AddressFamily == AddressFamily.InterNetwork)
```

Comment:
Hmm, this doesn't seem to be in master for some reason

Reply:
I noticed that too, note that is merging to 3.0.8 👍